src/website/apps/blog/views.py

```
@@ -0,0 +1,55 @@
+    ),
+    retrieve=extend_schema(
+        summary='Retrieve a blog post',
+        description='Returns a single published blog post using the public identifier or legacy numeric ID.',
+    ),
+)
+class BlogPostViewSet(SlugModelViewSetMixin, viewsets.ReadOnlyModelViewSet):
+    queryset = BlogPost.objects.all()
+    permission_classes = [IsAuthenticatedOrReadOnly]
+    filter_backends = [OrderingFilter]
+    ordering_fields = ['order', 'published_at', 'title', 'created']
+    ordering = ['order', '-published_at']
```

Ruff (0.15.12)

[warning] Mutable default value for class attribute

(RUF012)